beacon-chain/rpc/beacon/validators.go

```diff
@@ -492,10 +492,18 @@ func (bs *Server) GetValidatorParticipation(
 	}
 
 	// Get current slot state for current epoch attestations.
-	state, err := bs.StateGen.StateBySlot(ctx, currentSlot)
+	startSlot, err := helpers.StartSlot(requestedEpoch)
+	if err != nil {
+		return nil, err
+	}
+	// Use the last slot of requested epoch to obtain current and previous epoch attestations.
+	// This ensures that we don't miss previous attestations when input requested epochs.
+	startSlot += params.BeaconConfig().SlotsPerEpoch - 1
```

Comment:
what if im running a testnet where slotsperepoch is 1?

Reply:
then start slot would be epoch start slot and that's ok.

Reply:
makes sense!